runtime/src/utxo.rs

```diff
@@ -144,199 +100,152 @@ decl_module! {
-                salt: <system::Module<T>>::block_number().saturated_into::<u64>(),
             };
 
-            let hash = BlakeTwo256::hash_of(&utxo);
+            let hash = BlakeTwo256::hash_of(&(&utxo, 
+                        <system::Module<T>>::block_number().saturated_into::<u64>()));
 
-            if !<UnspentOutputs>::exists(hash) {
-                <UnspentOutputs>::insert(hash, utxo);
-                sp_runtime::print("leftover share sent to");
+            if !<UtxoStore>::exists(hash) {
+                <UtxoStore>::insert(hash, utxo);
+                sp_runtime::print("transaction reward sent to");
                 sp_runtime::print(hash.as_fixed_bytes() as &[u8]);
             } else {
-                sp_runtime::print("leftover share wasted due to hash collision");
+                sp_runtime::print("transaction reward wasted due to hash collision");
             }
         }
     }
-
-    /// Update storage to reflect changes made by transaction
-    fn update_storage(transaction: &Transaction, leftover: Value) -> DispatchResult {
-        // Calculate new leftover total
-        let new_total = <LeftoverTotal>::get()
-            .checked_add(leftover)
-            .ok_or("Leftover overflow")?;
-        <LeftoverTotal>::put(new_total);
-
-        // Storing updated leftover value
-        for input in &transaction.inputs {
-            <UnspentOutputs>::remove(input.parent_output);
+        
+    // Strips a transaction of its Signature fields by replacing value with ZERO-initialized fixed hash.
+    pub fn get_simple_transaction(_transaction: &Transaction) -> Vec<u8> {//&'a [u8] {
+        let mut trx = _transaction.clone();
+        for input in trx.inputs.iter_mut() {
+            input.sigscript = H512::zero();
         }
 
-        // Add new UTXO to be used by future transactions
-        for output in &transaction.outputs {
-            let hash = BlakeTwo256::hash_of(output);
-            <UnspentOutputs>::insert(hash, output);
-        }
-
-        Ok(())
+        trx.encode()
     }
 
-    pub fn lock_utxo(hash: &H256, until: Option<T::BlockNumber>) -> DispatchResult {
-        ensure!(!<LockedOutputs<T>>::exists(hash), "utxo is already locked");
-        ensure!(<UnspentOutputs>::exists(hash), "utxo does not exist");
-
-        if let Some(until) = until {
-            ensure!(
-                until > <system::Module<T>>::block_number(),
-                "block number is in the past"
-            );
-            <LockedOutputs<T>>::insert(hash, LockStatus::LockedUntil(until));
-        } else {
-            <LockedOutputs<T>>::insert(hash, LockStatus::Locked);
+    /// Helper fn for Transaction Pool
+    /// Checks for race condition, if a certain trx is missing input_utxos in UtxoStore
+    /// If None missing inputs: no race condition, gtg
+    /// if Some(missing inputs): there are missing variables
+    pub fn has_race_condition(_transaction: &Transaction) -> Option<Vec<&H256>> {
```

Comment:
The `Option<_>` seems unnecessary. Why not just return `Vec<&H256>`, and return the empty `Vec` when there is no race condition.

This is just a style comment, so take it or leave it.

Reply:
I like to check for `None`. leaving it :P